### PR TITLE
Forms: Add response inbox filters

### PR DESCRIPTION
**Pull request by jcheringer**
## Proposed changes:

* Adds the Date and Source dropdown filters to the Responses inbox view

image

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:

Enable the jetpack forms package and the new dashboard before testing with:

```
add_filter( 'jetpack_contact_form_use_package', '__return_true' );
add_filter( 'jetpack_forms_dashboard_enable', '__return_true' );
```

- Go to `/wp-admin/admin.php?page=jetpack-forms`
- Check if the "Date" and "Source" filters work properly

### CODE REVIEW

**Review by digitalwaveride — DISMISSED** [review]
Awesome João!!!
Works as advertised!
🚀

File: projects/packages/forms/src/dashboard/inbox/util.js [hunk 1]
```diff
@@ -10,6 +10,15 @@ export const getDisplayName = response => {
 	return response.ip;
 };
 
+export const getMonthName = monthNumber => {
+	const date = new Date();
+	date.setMonth( monthNumber - 1 );
+
+	return date.toLocaleString( 'en-US', {
```
Comment (ice9js) [inline projects/packages/forms/src/dashboard/inbox/util.js]: I think you probably want to use [dateI18n](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-date/#datei18n) here or even as a replacement for this function providing compatibility for all locales.

**Review by ice9js — DISMISSED** [review]
Looks good, works well. Just left one comment regarding the month dropdown formatting. I think it'd be preferable if we can localize that too.

File: projects/packages/forms/src/dashboard/inbox/util.js (hunk 1 above)
Comment (jcheringer) [reply projects/packages/forms/src/dashboard/inbox/util.js]: Thanks for pointing that out, @ice9js.
My latest commit aims to fix this.
Let me know if you thought of anything different or if I can improve anything else.

**Review by digitalwaveride — APPROVED** [review]
After adding the localization ... still works! 😉 🚀

**Pull request merged**